### PR TITLE
fix: [PL-14961]: Prevent Harness SM edit via YAML

### PR DESCRIPTION
**Pull request by tiwaritejas**
##### Summary:

Disallow Harness SM edit via YAML

##### Jira Links:

https://harness.atlassian.net/browse/PL-14961

##### Screenshots:

Screenshot 2022-06-02 at 2 15 52 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by tiwaritejas** [comment]
retrigger titlecheck

**Comment by tiwaritejas** [comment]
retrigger titlecheck

File: src/modules/10-common/components/YAMLBuilder/YamlBuilder.tsx [hunk 1]
```diff
@@ -388,10 +382,10 @@ const YAMLBuilder: React.FC<YamlBuilderProps> = (props: YamlBuilderProps): JSX.E
   })
 
   const handleEditorKeyDownEvent = (event: IKeyboardEvent, editor: any): void => {
-    if (props.isReadOnlyMode && isEditModeSupported) {
+    if (props.isReadOnlyMode && isEditModeSupported && !isHarnessManaged) {
       openDialog()
-    } else if (props.isReadOnlyMode && !isEditModeSupported && !hideErrorMesageOnReadOnlyMode) {
-      showNoPermissionError()
+    } else {
+      showYamlError?.()
```
Comment (avinashmadhwani02) [inline src/modules/10-common/components/YAMLBuilder/YamlBuilder.tsx:388]: Should we rename `showYamlError`? are we calling this when there's an error?

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/10-common/components/YAMLBuilder/YamlBuilder.tsx (hunk 1 above)
Comment (tiwaritejas) [reply src/modules/10-common/components/YAMLBuilder/YamlBuilder.tsx:388]: Yes, we call this when we want to display an error modal on UI. However, I am refactoring this due to some design issues. I will let you know once the changes are in place

**Review by meenakshi-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**